From db650a86b7dd860e6942e9e21b85f0bcfdc18137 Mon Sep 17 00:00:00 2001
From: Joseph Fernandes <josferna@redhat.com>
Date: Thu, 26 Nov 2015 14:33:14 +0530
Subject: [PATCH 175/178] tier/ctr: Check filename in ctr_lookup for nameless lookup

Check filename in ctr_lookup for nameless lookup

Backport of http://review.gluster.org/12760

> Change-Id: I1ab3f658e3b9fee708f6986d1990f16db920d2fb
> BUG: 1285663
> Signed-off-by: Joseph Fernandes <josferna@redhat.com>
> Reviewed-on: http://review.gluster.org/12760
> Tested-by: Gluster Build System <jenkins@build.gluster.com>
> Reviewed-by: Dan Lambright <dlambrig@redhat.com>
> Tested-by: Dan Lambright <dlambrig@redhat.com>
Signed-off-by: Joseph Fernandes <josferna@redhat.com>

Change-Id: I10006bf56fc95e3ef82be3d19d31f3e686b21775
BUG: 1284834
Reviewed-on: https://code.engineering.redhat.com/gerrit/63434
Reviewed-by: Joseph Fernandes <josferna@redhat.com>
Tested-by: Joseph Fernandes <josferna@redhat.com>
---
 .../changetimerecorder/src/changetimerecorder.c    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xlators/features/changetimerecorder/src/changetimerecorder.c b/xlators/features/changetimerecorder/src/changetimerecorder.c
index a230b2f..04cd314 100644
--- a/xlators/features/changetimerecorder/src/changetimerecorder.c
+++ b/xlators/features/changetimerecorder/src/changetimerecorder.c
@@ -329,7 +329,7 @@ ctr_lookup (call_frame_t *frame, xlator_t *this,
         GF_ASSERT(frame->root);
 
         /* Dont handle nameless lookups*/
-        if (!loc->parent)
+        if (!loc->parent || !loc->name)
                 goto out;
 
         /*fill ctr link context*/
-- 
1.7.1

